pkg/publish/default.go

```diff
@@ -203,7 +207,13 @@ func (d *defalt) Publish(ctx context.Context, br build.Result, s string) (name.R
 	// https://github.com/google/go-containerregistry/issues/212
 	s = strings.ToLower(s)
 
-	ro := []remote.Option{remote.WithAuth(d.auth), remote.WithTransport(d.t), remote.WithContext(ctx), remote.WithUserAgent(d.userAgent)}
+	ro := []remote.Option{remote.WithTransport(d.t), remote.WithContext(ctx), remote.WithUserAgent(d.userAgent)}
+	if d.auth == authn.Anonymous {
```

Comment:
```suggestion
	if d.auth != authn.Anonymous {
```

Right?

Maybe we should only store a `keychain`, and if someone calls `WithAuth` we should set `d.keychain` to a keychain that only returns that auth. Then we don't have to handle a possible situation where we store both an auth and a keychain, and have to decide which one to use.

Reply:
>Maybe we should only store a keychain
That's how I did it first. Do you want me to revert to the first version of the development?

^Based on that, if I understand correctly, we should remove the if logic and use the keychain only.

Right?

Reply:
Yeah, this struct would only hold a `keychain`, either the exactly the one passed by `WithAuthFromKeychain`, or a new keychain impl that always returns the auth passed by `WithAuth`.

Reply:
>this struct would only hold a keychain, either the exactly the one passed by WithAuthFromKeychain

I would like to go with this, wdyt?